### DISCUSSION TITLE
Import fuction not taking

### DISCUSSION THREAD

**Discussion by martienne17** [AprilSylph/XKit-Rewritten#1304] [Help Desk] [answered]
I have used the Import function before so I know how, but today it's not responding when I hit the Restore button at the bottom. I have so many tag bundles I have 4 pagedowns, so I'd really like to be able to restore them.

ETA: I left out a bunch of stuff I meant to include, excuse me. I just rebooted and it didn't fix anything. i use firefox on linux mint 21.1. I'm on the latest kernal and latest version of everything.

**Comment by marcustyphoon** [comment]
Ah, we don't have any UI that displays when the import function fails to parse the text you pasted. We should make one.

Did you accidentally forget to copy a character, perhaps the `{` at the beginning of your saved backup or the `}` at the end? (You can use a code editor or an online tool to see if there's a problem with a block of text that needs to be parsed as JSON; I did a quick web search and <https://codebeautify.org/jsonvalidator> seems to work fine for this.)

**Reply by martienne17** [reply]
I used ctrl+A in the text editor to select the text.

**Reply by marcustyphoon** [reply]
If you paste the text into the tool I linked, does it tell you that it's invalid?

**Reply by martienne17** [reply]
Chronic illness, sorry, just got the energy to try this. I got this result:
![image](https://github.com/AprilSylph/XKit-Rewritten/assets/142548371/4307203c-e350-4d1b-8a0e-91fd881966e0)
However, I don't see that error in the line:
```    {
      "title": "— mashup",
      "tags": "mashup, mashups, music, music poll, video, polls"
    },
    {
      "title": "🎖️ cover tournament",
      "tags": "tournament poll, polls, music, music poll, cover, covers, music cover tournament, video"
    },

**Reply by marcustyphoon** [reply, marked as answer]
Hm, that's strange! Your screenshot has `} {` without a comma between them, which would indeed cause that error, but your quoted block has the comma in that spot! Not sure what's happening there.

**Reply by martienne17** [reply]
I had to revalidate and fix more than one mistake. Thank you for helping.

**Comment by martienne17** [comment]
Wait, I think I fixed it and then didn't revalidate. But pasting it in
still doesn't work? Let me check that, one sec.
